endpoint/endpoint.go

```diff
@@ -107,7 +108,21 @@ func (t Targets) IsLess(o Targets) bool {
 
 	for i, e := range t {
 		if e != o[i] {
-			return e < o[i]
+			// Explicitly prefers IP addresses (e.g. A records) over FQDNs (e.g. CNAMEs).
+			// This prevents behavior like `1-2-3-4.example.com` being "less" than `1.2.3.4` when doing lexicographical string comparison.
+			ipA, _ := netip.ParseAddr(e)
```

Comment:
What happens if `ParseAddr` returns an error? Can we guarantee that it is impossible and thus ignore the error? If this is the case, can we add an error to clarify it to future readers? Same for the line right below this one.

Reply:
@Raffo Any parsing errors are caught when the `IsValid()` function is run below, since [an invalid IP address is returned](https://pkg.go.dev/net/netip#Addr) in those cases.

I will add debug-level logging to print out the error and a note to clarify, though.